### ISSUE TITLE
Summary: Customize/set title for `aria-label` on edit button

### ISSUE THREAD

**Issue by olemartinorg** [Altinn/app-frontend-react#1188]
### Description

We already set a `aria-label` when we can, but there should be more flexibility:
- We should probably support a `title` text resource binding on the `Summary` component as well, and display it as a header above the component. This can be achieved today by adding the `Summary` component to a `Group` and setting a `title` on the group (see the referenced Slack thread for an example). I'm assuming a `title` on a `Summary` component does nothing today, so one problem with adding support for this is the 5 apps I found that have configured a `title` binding on a Summary component (most notably `dsb/melding-om-sikkerthetsraadgiver` in prod).
- If nothing else works, we should have an optional binding for setting the `aria-label` for the edit button specifically. Might as well add a third binding for controlling the edit button text as well. 

### Additional Information

- #1100 
- #1102 
- https://altinn.slack.com/archives/C04CZ3KND6Z/p1684914915190359
